src/Aspire.Dashboard/Model/ValidateTokenMiddleware.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Claims;
+using Aspire.Dashboard.Configuration;
+using Aspire.Dashboard.Utils;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Options;
+using System.Web;
+
+namespace Aspire.Dashboard.Model;
+
+internal sealed class ValidateTokenMiddleware
```

Comment:
Curious as to why this is a middleware and not just an authentication handler?

Reply:
Can you log an issue to follow up later? I think we should be targeting using the framework primitives to implement this and if we wanted it as a new viable standalone authentication option for ASP.NET Core apps (as I know @davidfowl has mentioned) that would be a requirement.